# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

(a + b) * c / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [2]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [3]:
11283 // 89

126

4. Запишіть у змінну `name` ваше імʼя.

In [5]:
name = 'Marta'

5. Запишіть у змінну `surname` ваше прізвище.

In [7]:
surname = 'Hvozd'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [11]:
name_surname = name + ' ' + surname
name_surname

'Marta Hvozd'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [12]:
name_surname.strip().upper()

'MARTA HVOZD'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [13]:
print(name_surname[2])

r


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [67]:
print(surname[0])
print(surname[-1])

H
d


In [68]:
print(f"Перша літера: {surname[0]}, остання літера: {surname[-1]}")

Перша літера: H, остання літера: d


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [69]:
len(surname)

5

In [19]:
print(f"Довжина прізвища: {len(surname)}")

Довжина прізвища: 5


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

Оскільки потрібно врахувати і великі, і малі літери A/a, ми переведемо спочатку рядок у нижній регістр за допомогою `.lower()` і потім порахуємо кількість малих 'a'.

In [23]:
Madagaskar_capital = 'Antananarivo'

print(Madagaskar_capital.lower().count("a"))

4


Очікувана відповідь: 4

Наступні вправи можуть здатись дещо складніше - вони для найуважніших!
Якщо не виходить - не страшно. Пишіть нам в чат, ми підкажемо!

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

Основна відповідь:

In [30]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

True
False


Допоміжні тренувальні розрахунки для розуміння індексації:

In [39]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

print(my_string[0])
print(my_string[:1])
print(my_string[:2])
print(my_string[:3])
print(my_string[:4])
print(my_string[:5])

print(my_string[-1:])
print(my_string[-2:])
print(my_string[-3:])
print(my_string[-4:])

T
T
Th
The
The 
The q
!
!!
!!!
g!!!


Допоміжні тренувальні розрахунки для розуміння індексації:

In [52]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'

print(my_string[:3] == 'The')
print(my_string[-4:] == 'dog!')

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

Відповідь:

In [60]:
print(my_string.split())

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']


In [59]:
print("_".join(my_string.split()))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Допоміжні тренувальні розрахунки:

In [58]:
words_my_string = my_string.split()

print(*words_my_string, sep="\n")

The
quick
brown
fox
jumps
over
the
lazy
dog!!!


In [57]:
final_my_string = "_".join(words_my_string)

print(final_my_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`

In [61]:
name = "Іван Петрович"

print('Привіт, ' + name)

Привіт, Іван Петрович


In [62]:
print('Привіт,', name)

Привіт, Іван Петрович


# Тести
Перевіряємо чи всі букви тексту в нижньому регістрі:

In [64]:
text = "python"

print(text.islower())

True


In [65]:
text = "Python"

print(text.islower())

False
